frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -53,18 +67,67 @@ const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSource
     [setErrors, setStatus, setFieldValue, setFieldTouched, selectedKey, validateForm],
   );
 
-  const itemSelectorField = (
-    <ItemSelectorField
-      itemList={eventSourceList}
-      loadingItems={!eventSourceItems}
-      name="type"
-      onSelect={handleItemChange}
-      autoSelect
-    />
+  const filterSources = React.useCallback(
+    (text: string) => {
+      setFilteredSources(
+        _.pickBy(eventSourceList, (_val, key) => fuzzy(text.toLowerCase(), key.toLowerCase())),
+      );
+    },
+    [eventSourceList],
+  );
+
+  const debounceFilterText = useDebounceCallback<(text: string) => void>(filterSources, [
+    filterSources,
+  ]);
+
+  const handleFilterChange = (text: string) => {
+    setFilterText(text);
+    debounceFilterText(text);
+  };
+
+  const textFilter = (
```

Comment:
Would be better to use this filter inside `ItemSelectorField` so that it is reusable if other forms want the selector with a filter.

Reply:
Moved filter to `ItemSelectorField`, can now be enabled through `showFilter` and `showCount` props.